tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/BaseOnboard.kt

```diff
@@ -118,51 +130,53 @@ abstract class BaseOnboard : Runnable {
-    }
-
-    private val url by lazy {
-        urlFromClusterName(cordaClusterName, restWorkerDeploymentName)
-    }
-
-    protected fun setupClient() {
-        Unirest.config()
-            .verifySsl(false)
-            .setDefaultBasicAuth("admin", restPassword)
-            .defaultBaseUrl(url)
-    }
-
     internal class OnboardException(message: String) : Exception(message)
 
-    internal fun <T> HttpResponse<T>.bodyOrThrow(): T {
-        if (!this.isSuccess) {
-            throw OnboardException("Error: ${this.body}")
-        }
-        return this.body
-    }
     protected fun uploadCpi(cpi: InputStream, name: String): String {
-        val id = cpi.use { jarInputStream ->
+        createRestClient(CpiUploadRestResource::class).use { client ->
+            cpi.use { jarInputStream ->
+                client.start().also { connection ->
+                    val resource = connection.proxy
+                    val upload = HttpFileUpload(content = jarInputStream, fileName = "$name.cpi")
+                    val id = resource.cpi(upload).id
+
+                    // Wait for some time before checking the status of the CPI upload.
+                    Thread.sleep(2000)  // You can adjust this delay as per your requirements.
+
+                    // Retry for a certain number of times if the check fails
+                    var retryCount = 0
+                    val maxRetryCount = 3
+
+                    while (retryCount < maxRetryCount) {
+                        try {
+                            return checkCpiStatus(id)
+                        } catch (e: OnboardException) {
+                            println("Check CPI status failed for id: $id, retrying...")
+                            retryCount++
+                        }
+                    }
+
+                    // If all retries failed, throw the exception
+                    throw OnboardException("CPI request $id had failed after $maxRetryCount attempts!")
+                }
+            }
 
-            Unirest.post("/cpi")
-                .field("upload", jarInputStream, "$name.cpi")
-                .asJson()
-                .bodyOrThrow()
-                .`object`.get("id")
         }
+    }
+
+    private fun checkCpiStatus(id: String): String {
         val end = System.currentTimeMillis() + 5 * 60 * 1000
         while (System.currentTimeMillis() < end) {
-            val response = Unirest.get("/cpi/status/$id").asJson()
-            val status = try {
-                response.bodyOrThrow().`object`.get("status")
-            } catch (e: OnboardException) {
-                "Not ready yet"
-            }
-            if (status == "OK") {
-                return response.body.`object`.get("cpiFileChecksum").toString()
+            createRestClient(CpiUploadRestResource::class).use { client ->
+                client.start().also { connection ->
+                    val resource = connection.proxy
```

Comment:
We can probably do something like this everywhere to avoid the single-use vals and `also{ }` blocks. You could refer to the existing usages of `createRestClient` and `executeWithRetry` for consistency -
```suggestion
                createRestClient(Example::class).use { client ->
                val endpoint = executeWithRetry(waitDuration, "Sample operation name") {
                    client.start().proxy
                }
```